### PR TITLE
samples: openthread: ading rtt and logging files

### PR DESCRIPTION
**Pull request by canisLupus1313**
This commit adds rtt and logging overlays across all openthread
samples. This is done for consistency in build process and to
avoid hacks in test framework.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

### CODE REVIEW

File: samples/openthread/coprocessor/overlay-rtt.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Disable Spinel logging backend
```
Comment (edmont) [inline samples/openthread/coprocessor/overlay-rtt.conf:6]: Move this below RTT enabling for consistency with other files.

**Comment by edmont** [comment]
What about adding and entry to the changelog?

* Logging and RTT overlays available in all samples.

**Review by edmont — APPROVED** [review]

File: samples/openthread/coap_client/overlay-logging.conf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Enable Zephyr logging
+CONFIG_LOG=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_client/overlay-logging.conf:7]: no need for this here, logging is already enabled in prj.conf file

File: samples/openthread/coap_client/overlay-logging.conf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# Option for configuring log level in this sample
+CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL_INF=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_client/overlay-logging.conf:10]: remove it, it is for CLI sample only

File: samples/openthread/coap_server/overlay-logging.conf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Enable Zephyr logging
+CONFIG_LOG=y
+
+# Option for configuring log level in this sample
+CONFIG_OT_COMMAND_LINE_INTERFACE_LOG_LEVEL_INF=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_server/overlay-logging.conf:10]: remove it, it is for CLI sample only

File: samples/openthread/coap_server/overlay-logging.conf [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Enable Zephyr logging
+CONFIG_LOG=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_server/overlay-logging.conf:7]: no need for this here, logging is already enabled in prj.conf file

File: samples/openthread/coap_server/overlay-logging.conf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# CONFIG_NET_CONFIG_LOG_LEVEL_INF=y
+
+# Option for configuring log level in Zephyr L2 logging
+# CONFIG_OPENTHREAD_L2_DEBUG=y
+# CONFIG_OPENTHREAD_L2_LOG_LEVEL_DBG=y
+# CONFIG_OPENTHREAD_L2_DEBUG_DUMP_15_4=y
+# CONFIG_OPENTHREAD_L2_DEBUG_DUMP_IPV6=y
+
+# Enable OpenThread logging
+CONFIG_OPENTHREAD_DEBUG=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_server/overlay-logging.conf:24]: no need for this here, it is already enabled in prj.conf file

File: samples/openthread/coap_client/overlay-logging.conf [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+# CONFIG_NET_CONFIG_LOG_LEVEL_INF=y
+
+# Option for configuring log level in Zephyr L2 logging
+# CONFIG_OPENTHREAD_L2_DEBUG=y
+# CONFIG_OPENTHREAD_L2_LOG_LEVEL_DBG=y
+# CONFIG_OPENTHREAD_L2_DEBUG_DUMP_15_4=y
+# CONFIG_OPENTHREAD_L2_DEBUG_DUMP_IPV6=y
+
+# Enable OpenThread logging
+CONFIG_OPENTHREAD_DEBUG=y
```
Comment (lmaciejonczyk) [inline samples/openthread/coap_client/overlay-logging.conf:24]: no need for this here, it is already enabled in prj.conf file

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]

**Review by lmaciejonczyk — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**